src/cpu/x64/cpu_isa_traits.hpp

```diff
@@ -336,6 +350,27 @@ struct cpu_isa_traits_t<avx10_1_512_amx_fp16> {
     static constexpr const char *user_option_env = "avx10_1_512_amx_fp16";
 };
 
+template <typename Wmm>
+struct avx10_t {
```

Comment:
Should this inherit `vreg_traits_t`? If not, it probably should contain all its members to be usable/functional.